vault/auth.go

```diff
@@ -499,50 +500,210 @@ func (c *Core) taintCredEntry(ctx context.Context, nsID, path string, updateStor
 
 // loadCredentials is invoked as part of postUnseal to load the auth table
 func (c *Core) loadCredentials(ctx context.Context) error {
-	// Load the existing mount table
-	raw, err := c.barrier.Get(ctx, coreAuthConfigPath)
+	// Previously, this lock would be held after attempting to read the
```

Comment:
Hmm, `loadCredentials` is only called during unseal, which is strictly linear anyways and does not allow handling of requests yet, so I actually don't think this lock is necessary, much less performance sensitive. But while I'm inclined to leave it in case we later restructure unseal... I don't think the refactor into a `WithLock`/`WithoutLock` makes sense to do? 

Thoughts?

Reply:
Actually, with the above arguments around this only happening at unseal, I see the linearity case your are making. In that case, this is ok.